src/awscdk/github.ts

```diff
@@ -79,15 +94,18 @@ export class GithubCDKPipeline extends CDKPipeline {
       env: {
         CI: 'true',
       },
+      needs: [...preInstallSteps.flatMap(s => s.needs), ...preSynthSteps.flatMap(s => s.needs), ...postSynthSteps.flatMap(s => s.needs)],
       permissions: { idToken: JobPermission.WRITE, contents: JobPermission.READ },
       steps,
     });
   }
 
   public createAssetUpload(): void {
+    const preInstallSteps = (this.options.preInstallSteps ?? []).map(s => s.toGithub());
```

Comment:
should'nt this be more generic - or instead of doing .toGithub it should be something like .transform() for the step object and the step would then know where to transform it to?

Reply:
As we are in the GitHub implementation file I think calling toGithub is fine